### DISCUSSION TITLE
How to specify permission bits for $os.writeFile, $os.mkdirAll, etc?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#5513] [Q&A] [answered]
`  $os.writeFile(path, data)` seems to work without specifying any permissions, but I'm not sure how to pass an `os.FileMode` in. It looks like a `number` in `jsvm.d.ts`, is there an example?

**Comment by ganigeorgiev** [comment, marked as answer]
`$os.writeFile()` and `$os.mkdirAll()` are direct binding to the Go equivalent [`os.WriteFile()`](https://pkg.go.dev/os#WriteFile) and [`os.MkdirAll()`](https://pkg.go.dev/os#MkdirAll) functions.
We don't have the Go `os` and `fs` constants exposed to the JSVM but you can specify the permission bits directly for example like:

```js
$os.writeFile(path, data, 0644)
```

**Reply by einarpersson** [reply]
Has this changed?

![image](https://github.com/user-attachments/assets/33069002-9fd3-4f52-b40d-2dd11ae337d0)

It works however, despite the type error. Also consider adding this to the jsvm docs, for example [here](https://pocketbase.io/jsvm/interfaces/os.FileMode.html) with a concrete example.

**Reply by ganigeorgiev** [reply]
@einarpersson The JSVM reference docs is autogenerated from the Go source code.

Regarding the error - this seems like a LSP warning and you can either apply the suggestion or leave it as it is.
